wger/core/api/views.py

```diff
@@ -125,3 +137,56 @@ class WeightUnitViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = WeightUnitSerializer
     ordering_fields = "__all__"
     filter_fields = ("name",)
+
+
+class ApiCreateUser(viewsets.ModelViewSet):
+    permission_classes = (AllowAny,)
+    serializer_class = (ApiCreateUserSerializer)
+
+    def create(self, request):
+        token = request.data.get('token')
+        user_id_check = get_object_or_404(Token, key=token)
+        permission = UserProfile.objects.filter(
+            user=user_id_check.user_id, is_allowed=True)
+        if not permission:
+            return Response(
+                'You arent authorised to create user contact admin for rights',
+                status=status.HTTP_400_BAD_REQUEST)
+        serializer = self.serializer_class(data=request.data)
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        permission = UserProfile.objects.filter(
+            user=self.request.user.id, is_allowed=True)
+        if User.objects.filter(
+                username=serializer.validated_data['username']).exists():
+            return Response('user exists', status=status.HTTP_400_BAD_REQUEST)
+        if User.objects.filter(
+                email=serializer.validated_data['email']).exists():
+            return Response('user exists', status=status.HTTP_400_BAD_REQUEST)
+        user = User.objects.create_user(
+            username=serializer.data['username'],
+            email=serializer.data['email'])
+        user.save()
+        user = User.objects.get(pk=user.id)
+        # Pre-set some values of the user's profile
+        language = Language.objects.get(
+            short_name=translation.get_language()
+        )
+        user.userprofile.notification_language = language
+        # Set default gym, if needed
+        gym_config = GymConfig.objects.get(pk=1)
+        if gym_config.default_gym:
+            user.userprofile.gym = gym_config.default_gym
+            # Create gym user configuration object
+            config = GymUserConfig()
+            config.gym = gym_config.default_gym
+            config.user = user
+            config.save()
+        user.userprofile.save()
+        # update the APIUser table
+
+        new_user = ApiUserModel.objects.create(
+            user_being_created=user,
+            creator=User.objects.get(pk=user_id_check.user_id))
+        new_user.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
```

Comment:
This class has very many lines of code. Consider the refactoring suggested above